File: sample/todo-app-ios/todo-app-ios/LifecycleWrapper.swift
//
//  ViewLifecycle.swift
//  todo-app-ios
//
//  Created by stream on 4/29/20.
//  Copyright © 2020 arkivanov. All rights reserved.
//

import TodoLib

class LifecycleWrapper {

    let lifecycle = LifecycleRegistry()
    
    init() {
        self.lifecycle.onCreate()
    }
    
    deinit {
        self.lifecycle.destroy()
    }
    
    func start() {
        lifecycle.resume()
    }
    
    func stop() {
        lifecycle.stop()
    }
}

File: sample/todo-app-js/src/main/kotlin/root/LifecycleWrapper.kt
package root

import com.arkivanov.mvikotlin.core.lifecycle.LifecycleRegistry

class LifecycleWrapper {

    val lifecycle = LifecycleRegistry()

    init {
        debugLog("lifecycle onCreate")
        lifecycle.onCreate()
    }

    fun start() {
        debugLog("lifecycle onStart")
        lifecycle.onStart()
        debugLog("lifecycle onResume")
        lifecycle.onResume()
    }

    fun stop() {
        debugLog("lifecycle onPause")
        lifecycle.onPause()
        debugLog("lifecycle onStop")
        lifecycle.onStop()
        debugLog("lifecycle onDestroy")
        lifecycle.onDestroy()
    }
}